models/dictionary.go

```diff
@@ -19,7 +17,66 @@ type Word struct {
 	Link     string `json:"_"`
 }
 
-func MakeWord(c *html.Node, link, detail string, index int) *Word {
+type WonderWord struct {
+	Index       int    `json:"index"`
+	Term        string `json:"term"`
+	Reading     string `json:"reading"`
+	Explanation string `json:"explanation"`
+	Example     string `json:"example"`
+	Mean        string `json:"mean"`
+}
+
+type Data interface {
+	GetIdx() int
+	GetTextOrTerm() string
+	GetMean() string
+	SetMean(string)
+	GetDetail() string
+}
+
+func (w *Word) GetIdx() int {
+	return w.Index
+}
+
+func (w *Word) GetTextOrTerm() string {
+	return w.Text
+}
+
+func (w *Word) GetMean() string {
+	return w.Text
+}
+
+func (w *Word) SetMean(mean string) {
+	w.MeanEng = mean
+}
+
+func (w *Word) GetDetail() string {
+	return w.Text
+}
+
+func (w *WonderWord) GetIdx() int {
+	return w.Index
+}
+
+func (w *WonderWord) GetTextOrTerm() string {
+	return w.Term
+}
+
+func (w *WonderWord) GetMean() string {
+	return w.Mean
+}
+
+func (w *WonderWord) SetMean(mean string) {
+	w.Mean = mean
+}
+
+func (w *WonderWord) GetDetail() string {
+	return w.Explanation
+}
+
+type MakeData func(c *html.Node, idx int, option ...string) Data
+
+func MakeWord(c *html.Node, index int, options ...string) Data {
```

Comment:
check length của `options` trước khi access để tránh lỗi panic `index out of range`